### PR TITLE
Cache params ordering issue

### LINKED ISSUE #156: Check if `params` of the Query is serizlized before `key` generation
Some tech debt was introduced in https://github.com/igorkamyshev/farfetched/pull/131 🙃

Let's imagine:

```ts
const query = createQuery({ handler: (p) => null })

cache(query)

const someUnserializableData = /* ... */

query.start(someUnserializableData)
```

`JSON.stringify(someUnserializableData)` would be used as a part of a key, in some cases it could be unusful.

### PR DESCRIPTION
**Pull request by Minhir**
In the first commit, I represented the current problem with cache. Now, query params `{a: 1, b: 2}` and `{b: 2, a: 1}` are resolved as different ones.

In the second commit, I solved the problem with [fast-json-stable-stringify](https://www.npmjs.com/package/fast-json-stable-stringify) library as an example. I'm opened to discussion: maybe we will find another solution or hardcode stable-json-hash inside core package.

fixes #156

### CODE REVIEW

**Review by igorkamyshev — CHANGES_REQUESTED** [review]
Great idea 💙

However, I want to keep the core package with only one dependency — effector. So, I guess we can reimplement `fast-json-stable-stringify` inside package.

fixes #156

File: packages/core/src/cache/lib/stable_stringify.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+export function stableStringify(data: unknown): string | void {
```
Comment (igorkamyshev) [inline packages/core/src/cache/lib/stable_stringify.ts:1]: 🤓 а чому void?

**Review by igorkamyshev — APPROVED** [review]

File: packages/core/src/cache/lib/stable_stringify.ts (hunk 1 above)
Comment (Minhir) [reply packages/core/src/cache/lib/stable_stringify.ts:1]:
```javascript
JSON.stringify(undefined) // undefined
```
:man_shrugging: 

https://evanhahn.com/when-stringify-doesnt-return-a-string/

https://github.com/microsoft/TypeScript/issues/18879 -- even in TS it's an issue

**Comment by Minhir** [comment]
Will close https://github.com/igorkamyshev/farfetched/issues/156

File: .changeset/brown-colts-refuse.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@farfetched/core': minor
```
Comment (igorkamyshev) [inline .changeset/brown-colts-refuse.md:2]:
```suggestion
'@farfetched/core': patch
```

**Review by igorkamyshev — APPROVED** [review]
Let's add changeset about fixes bug with un-serializable params? After that, we can ,erge it.

**Pull request merged**